### ISSUE TITLE
Status value does not change

### ISSUE THREAD

**Issue by xpatohr** [microsoftgraph/security-api-solutions#40]
# Issue _Status value does not change_

Having problems with the "status" field from the Microsoft Graph Security API.

_Example:_
GET /graph.microsoft.com/v1.0/security/alerts/03da5d48-a892-4ea6-9949-dc902283e2d8f
Returns status 200 with a JSON object as below.

``` json
{
  "activityGroupName": "String",
  "assignedTo": "String",
  "azureSubscriptionId": "String",
  "azureTenantId": "String",
  "category": "String",
  "closedDateTime": "String (timestamp)",
  "cloudAppStates": [{"@odata.type": "microsoft.graph.cloudAppSecurityState"}],
  "comments": ["String"],
  "confidence": 1024,
  "createdDateTime": "String (timestamp)",
  "description": "String",
  "detectionIds": ["String"],
  "eventDateTime": "String (timestamp)",
  "feedback": "@odata.type: microsoft.graph.alertFeedback",
  "fileStates": [{"@odata.type": "microsoft.graph.fileSecurityState"}],
  "hostStates": [{"@odata.type": "microsoft.graph.hostSecurityState"}],
  "id": "String (identifier)",
  "lastModifiedDateTime": "String (timestamp)",
  "malwareStates": [{"@odata.type": "microsoft.graph.malwareState"}],
  "networkConnections": [{"@odata.type": "microsoft.graph.networkConnection"}],
  "processes": [{"@odata.type": "microsoft.graph.process"}],
  "recommendedActions": ["String"],
  "registryKeyStates": [{"@odata.type": "microsoft.graph.registryKeyState"}],
  "severity": "@odata.type: microsoft.graph.alertSeverity",
  "sourceMaterials": ["String"],
  "status": "@odata.type: microsoft.graph.alertStatus",
  "tags": ["String"],
  "title": "String",
  "triggers": [{"@odata.type": "microsoft.graph.alertTrigger"}],
  "userStates": [{"@odata.type": "microsoft.graph.userSecurityState"}],
  "vendorInformation": {"@odata.type": "microsoft.graph.securityVendorInformation"},
  "vulnerabilityStates": [{"@odata.type": "microsoft.graph.vulnerabilityState"}]
}
```

The problem is that the alert is **"resolved"** in **Cloud App Security**. But the API returns **"status": "newAlert"** on the object. The return value shoud be **"status": "resolved"**.

# Additional Info

Properties from https://docs.microsoft.com/en-us/graph/api/resources/alert?view=graph-rest-1.0

status | alertStatus | Alert lifecycle status (stage). Possible values are: unknown, newAlert, inProgress, resolved. (supports update). Required.
-- | -- | --

# General

**Comment by chinguyen1** [comment]
@xpatohr , so the issue is that the alert status was updated to "resolved" in the MCAS portal first and then you made a Get alert by ID call to Graph Security API to retrieve that alert and noticed the status of the alert still shows "newAlert". Is that correct?

**Comment by xpatohr** [comment]
> @xpatohr , so the issue is that the alert status was updated to "resolved" in the MCAS portal first and then you made a Get alert by ID call to Graph Security API to retrieve that alert and noticed the status of the alert still shows "newAlert". Is that correct?

That's right. The status is incorrect whether I make a GET call by id or GET all alerts without id.

**Comment by Roughain** [comment]
Hi Chi,
this is probably the same issue I mentioned in our last call rgarding updates to the Alert or incident in MCAS oder MDATP in the Portal, which generate an additional Alert with status newAlert available via the Graph API. Instead, the alerts should have a status of newAlert.

**Comment by chinguyen1** [comment]
@Roughain, Hi Florian, I think you are right that this is the same issue with the alert status update with MCAS. I am working with MCAS on this issue and the other ones you mentioned in the email. Thank you for your patience.

**Comment by xpatohr** [comment]
Is there any update on this @chinguyen1? Or do you have any recommendation on what i can do?

**Comment by preetikr** [comment]
@xpatohr @Roughain - Patch / update support for Microsoft Graph Security integrated applications is there, so if you update an MCAS alert from one Graph Security integrated app, you should see the update reflected across the other Graph Security integrated app. Now, the MCAS portal itself is not a Graph Security integrated app, so there's a feature needed to ensure the MCAS portal can reflect the updates shared by any Graph Security integrated app. We are working on enabling this support and will circle back on timelines for this. 
That said, this support is already enabled for Azure Security Center and Microsoft Defender ATP portals (alert update in an Graph Security API app is reflected in ASC and defender ATP portals). Thanks, for your feedback on this.

**Comment by SVAjcuepper** [comment]
Hello, the issue hasnt been resolved yet, right?

If I edit the alerts in the security center, graph api does not recognise the changes. Please fix!

**Comment by createchange** [comment]
I have the opposite happening - I update a finding via Graph API to `resolved`, and it is not reflected in the UI.

I presume this has the same underlying issue - the Graph API and the UI are simply disconnected. If this is the case, how is this acceptable? It has been 3 years since the original report.
